src/resolve/resolved-dns-rr.c

```diff
@@ -1222,8 +1222,10 @@ int dns_resource_record_to_wire_format(DnsResourceRecord *rr, bool canonical) {
                 return 0;
 
         r = dns_packet_append_rr(&packet, rr, 0, &start, &rds);
-        if (r < 0)
+        if (r < 0) {
+                dns_packet_unref(&packet);
```

Comment:
maybe used `_cleanup_(dns_packet_unref)` on the declaration of the struct above? i.e. not `dns_packet_unrefp` but the one without the `p`, given the structure is allocated directly on the stack instead of being a pointer.

Reply:
Already addressed by #27783 I guess.